presto-tests/src/main/java/com/facebook/presto/tests/AbstractTestQueries.java

```diff
@@ -6256,6 +6256,8 @@ public void testMapSubset()
         assertQuery("select m[1], m[3] from (select map_subset(map(array[1,2,3,4], array['a', 'b', 'c', 'd']), array[1,3,10]) m)", "select 'a', 'c'");
         assertQuery("select cardinality(map_subset(map(array[1,2,3,4], array['a', 'b', 'c', 'd']), array[10,20]))", "select 0");
         assertQuery("select cardinality(map_subset(map(), array[10,20]))", "select 0");
```

Comment:
I am actually wondering why this test was not causing hang before?

Reply:
Cardinality works lol - we have some strange map optimizations where certain things are expected to be of certain type. In particular MAP() is MAP<UNKNOWN, UNKNOWN> and that type cannot be created manually (it;s just inferred) So we need to return empty as they are.